code/game/objects/structures/flora.dm

```diff
@@ -1,83 +1,413 @@
+	/// If set, the flora will have this as its description after being harvested. When the flora becomes harvestable again, it regerts to its initial(desc)
+	var/harvested_desc
+
+	/// A lazylist of products that could be created when harvesting this flora, syntax is (type = weight)
+	/// Because of how this works, it can spawn in anomalies if you want it to. Or wall girders
+	var/product_types
+	/// A lazylist typecache of items that can harvest this flora.
+	/// Will be set automatically on Initialization depending on flora_flags.
+	/// Paths in this list and their subtypes will be able to harvest the flora.
+	var/required_tools
+	/// A lazylist typecache of items that will be excluded from required_tools
+	/// Paths in this list disallows items from harvesting this flora if that item is a type of this path
+	var/disallowed_tools
+	/// If the user is able to harvest this with their hands
+	var/harvest_with_hands = FALSE
+	/// The "verb" to use when the user harvests the flora
+	var/harvest_verb = "harvest"
+	/// What should be added to harvest_verb depending on the context ("user harvest(s) the tree" / "user chop(s down) the tree")
+	var/harvest_verb_suffix = "s"
+	/// If the user is allowed to uproot the flora
+	var/can_uproot = TRUE
+	/// What tools are allowed to be used to uproot the flora
+	var/uprooting_tools = list(/obj/item/shovel)
+	var/uprooted = FALSE
+	var/previous_rotation = 0
+	
+	/// If false, the flora won't be able to be harvested at all. If it's true, go through checks normally to determine if the flora is able to be harvested
+	var/harvestable = TRUE
+	/// The low end of how many product_type items you get
+	var/harvest_amount_low = 1
+	/// The high end of how many product_type items you get
+	var/harvest_amount_high = 3
+
+	//Messages to show to the user depending on how many items they get when harvesting the flora
+	var/harvest_message_low
+	var/harvest_message_med
+	var/harvest_message_high
+	/// See proc/harvest() and where this is used for an explaination on how this works
+	var/harvest_message_true_thresholds = FALSE
+
+	/// How long it takes to harvest the flora with the correct tool
+	var/harvest_time = 6 SECONDS
+	var/delete_on_harvest = FALSE
+	var/harvested = FALSE
+
+	/// Variables for determining the low/high ends of how long it takes for the flora takes to grow.
+	var/regrowth_time_low = 8 MINUTES
+	/// Stops the flora from regrowing if this is set to 0
+	var/regrowth_time_high = 16 MINUTES
+
+	/// Flags for the flora to determine what kind of sound to play when it gets hit
+	var/flora_flags = NONE
+
+/obj/structure/flora/Initialize(mapload)
+	. = ..()
+	if(!required_tools)
+		required_tools = list()
+		if(flora_flags & FLORA_WOODEN)
+			required_tools += FLORA_HARVEST_WOOD_TOOLS
+		if(flora_flags & FLORA_STONE)
+			required_tools += FLORA_HARVEST_STONE_TOOLS
+	
+	required_tools = typecacheof(required_tools)
+	disallowed_tools = typecacheof(disallowed_tools)
+	uprooting_tools = typecacheof(uprooting_tools)
```

Comment:
use a global list for these so you only have to do this once for each unique subtype please, id rather not deal with this type of performance problem again